wrapper/src/test/java/software/amazon/jdbc/plugin/DataCacheConnectionPluginTest.java

```diff
@@ -0,0 +1,123 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package software.amazon.jdbc.plugin;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.mockito.Mockito.when;
+
+import java.sql.ResultSet;
+import java.sql.ResultSetMetaData;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.Properties;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+
+class DataCacheConnectionPluginTest {
+
+  private static final Properties props = new Properties();
+
+  private AutoCloseable closeable;
+
+  @Mock
+  ResultSet mockResult1;
+  @Mock
+  ResultSet mockResult2;
+  @Mock
+  Statement mockStatement;
+  @Mock
+  ResultSetMetaData mockMetaData;
+
+
+  @BeforeEach
+  void setUp() throws SQLException {
+    closeable = MockitoAnnotations.openMocks(this);
+    props.setProperty(DataCacheConnectionPlugin.DATA_CACHE_TRIGGER_CONDITION.name, "foo");
+    DataCacheConnectionPlugin.clearCache();
+
+    when(mockResult1.getMetaData()).thenReturn(mockMetaData);
+    when(mockResult2.getMetaData()).thenReturn(mockMetaData);
+    when(mockMetaData.getColumnCount()).thenReturn(1);
+    when(mockMetaData.getColumnName(1)).thenReturn("fooName");
+
+    // Mock result sets contain 1 row.
+    when(mockResult1.next()).thenReturn(true, false);
+    when(mockResult2.next()).thenReturn(true, false);
+    when(mockResult1.getObject(1)).thenReturn("bar1");
+    when(mockResult2.getObject(1)).thenReturn("bar2");
+  }
+
+  @AfterEach
+  void cleanUp() throws Exception {
+    closeable.close();
+  }
+
+  @Test
+  void test_execute_withEmptyCache() throws SQLException {
+    final String methodName = "Statement.executeQuery";
+
+    final DataCacheConnectionPlugin plugin = new DataCacheConnectionPlugin(props);
+
+    final ResultSet rs = plugin.execute(
+        ResultSet.class,
+        SQLException.class,
+        mockStatement, methodName,
+        () -> mockResult1,
+        new String[]{"foo"}
+    );
+
+    compareResults(mockResult1, rs);
+  }
+
+  @Test
+  void test_execute_withCache() throws SQLException {
+    final String methodName = "Statement.executeQuery";
+
+    final DataCacheConnectionPlugin plugin = new DataCacheConnectionPlugin(props);
+
+    ResultSet rs = plugin.execute(
+        ResultSet.class,
+        SQLException.class,
+        mockStatement, methodName,
+        () -> mockResult1,
+        new String[]{"foo"}
+    );
+    compareResults(mockResult1, rs);
+
+    // Execute the query again with a different result set.
+    rs = plugin.execute(
+        ResultSet.class,
+        SQLException.class,
+        mockStatement, methodName,
+        () -> mockResult2,
+        new String[]{"foo"}
+    );
+
+    compareResults(mockResult1, rs);
```

Comment:
should we be verifying somehow that jdbcMethodFunc was only actually called once because the result set was returned from the cache the second time?